### PR TITLE
Updated to support .NET Core 3.1

### PR DESCRIPTION
**Pull request by petertuton**

### CODE REVIEW

**Comment by petertuton** [comment]
Also fixes #51 , updating to Azure Functions 3.0

**Comment by konichi3** [comment]
Thanks @petertuton . Can you add @marianan and @mhshami01 to reviewer?

**Comment by petertuton** [comment]
@konichi3, I'm unable to add reviewers, being neither a repository owner nor collaborator with write access to the repository.

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
     <TreatSpecificWarningsAsErrors />
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EdgeHub" Version="1.0.7" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
-    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.38" />
+    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.13" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.1.1" />
```
Comment (marianan) [inline content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:14]: >Microsoft.Extensions.Logging.Abstractions [](http://example.com/codeflow?start=31&length=41)

Why the change of reference from Microsoft.Extensions.Logging.Console to Microsoft.Extensions.Logging.Abstractions?

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
+    <PackageReference Include="Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator" Version="1.2.3" />
```
Comment (marianan) [inline content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:15]: >Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator [](http://example.com/codeflow?start=31&length=58)

What APIs does the project need from this reference?

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj (hunk 1 above)
Comment (petertuton) [reply content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:14]: In checking why this was needed, I can no longer determine the reason... I reset my dev environment to .NET 3.1.415 and everything appears fine when using `Microsoft.Extensions.Logging.Console`.

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj (hunk 2 above)
Comment (petertuton) [reply content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:15]: In checking why this was needed, I can no longer determine the reason... I reset my dev environment to .NET 3.1.415 and everything appears fine _without_ `Microsoft.Azure.WebJobs.Script.ExtensionsMetadataGenerator`.

**Comment by mhshami01** [comment]
Hello @petertuton, thank you for this contribution. Please validate your changes with the following steps:
image
Verify:
 - The contents of any newly created projects.
 - That you can build all platforms available (amd64, arm32v7...)

Sincererly,
--H

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj (hunk 2 above)
Comment (marianan) [reply content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:15]: that's great, let's remove this reference then if it is not needed.

File: content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj (hunk 1 above)
Comment (marianan) [reply content/dotnet-template-azure-iot-edge-function/CSharp/SampleFunction.csproj:14]: great, let's restore the original reference

**Comment by petertuton** [comment]
For completeness... 

Re `Microsoft.Azure.WebJobs.Script.MetadataGenerator`, it’s required to generate the `function.json` file for Azure Functions, as per [line 75](https://github.com/Azure/azure-functions-vs-build-sdk/blob/b0e54a832a92119e00a2b1796258fcf88e0d6109/src/Microsoft.NET.Sdk.Functions.MSBuild/Targets/Microsoft.NET.Sdk.Functions.targets#L75). However, nothing in the dependency tree loads this file, so an explicit include is required (perhaps an earlier version of the .NET SDK or Functions runtime included this file?). FYI, it wasn’t failing on my local machine because it was in my local machine’s `.nuget/packages` directory, but when building with the .NET 3.1 Core docker image, it was nowhere to be found.
 
Re `Microsoft.Extensions.Logging.Abstractions`, without its inclusion, I receive the following error when running the module in IoT Edge:
```
fail: Host.Startup[515]
      A host error has occurred during startup operation 'ce6635de-157b-445c-beb1-c0f511466a7f'.
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
```
